pkg/collector/smi_collector.go

```diff
@@ -0,0 +1,109 @@
+package collector
+
+import (
+	"log"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/aks-periscope/pkg/interfaces"
+	"github.com/Azure/aks-periscope/pkg/utils"
+)
+
+// SmiCollector defines an Smi Collector struct
+type SmiCollector struct {
+	BaseCollector
+}
+
+var _ interfaces.Collector = &SmiCollector{}
+
+// NewSmiCollector is a constructor
+func NewSmiCollector(exporter interfaces.Exporter) *SmiCollector {
+	return &SmiCollector{
+		BaseCollector: BaseCollector{
+			collectorType: Smi,
+			exporter:      exporter,
+		},
+	}
+}
+
+// Collect implements the interface method
+func (collector *SmiCollector) Collect() error {
+	// Get all CustomResourceDefinitions in the cluster
+	allCrdsList, err := utils.GetResourceList([]string{"get", "crds", "-o", "jsonpath={..metadata.name}"}, " ")
+	if err != nil {
+		return err
+	}
+
+	// Directory where logs will be written to
+	rootPath, err := utils.CreateCollectorDir(collector.GetName())
+	if err != nil {
+		return err
+	}
+
+	// Filter to obtain a list of Smi CustomResourceDefinitions in the cluster
+	crdNameContainsSmiPredicate := func(s string) bool { return strings.Contains(s, "smi-spec.io") }
+	smiCrdsList := filter(allCrdsList, crdNameContainsSmiPredicate)
+	if len(smiCrdsList) == 0 {
+		// TODO should this return an error?
```

Comment:
Should this return error here?

Reply:
Resolved in 5b56484 - this should be an error. If the user chooses to collect `SMI` data, then the scenario is that the user *has* `SMI` installed in their cluster, which is why they chose to collect `SMI` data. If they explicitly chose `SMI` data through the explicitly setting the flag, then this needs to return error if it fails to find any SMI CRDs in the cluster.